bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -501,7 +503,7 @@ public int getTimeSecond() {
      */
     @Deprecated
     public long getTime() {
-        return timestamp * 1000L;
+        return ((long)timestamp & 0xFFFFFFFFL) * 1000L;
```

Comment:
See above.  I think the cast is extraneous.

Reply:
Cast removed.